ranger-core/src/main/java/io/appform/ranger/core/model/ServiceRegistry.java

```diff
@@ -15,15 +15,30 @@
  */
 package io.appform.ranger.core.model;
 
-import lombok.Data;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 import java.util.List;
+import lombok.Getter;
 
-@Data
 public abstract class ServiceRegistry<T> {
+    @Getter
     private final Service service;
+    private final AtomicBoolean refreshed = new AtomicBoolean(false);
+
+    protected ServiceRegistry(Service service) {
+        this.service = service;
+    }
 
     public abstract List<ServiceNode<T>> nodeList();
 
-    public abstract void updateNodes(List<ServiceNode<T>> nodes);
+    protected abstract void update(List<ServiceNode<T>> nodes);
```

Comment:
Move protected functions to below public functions. Order should always be public, protected, private

Reply:
Thanks, made changes as per.